File: ADASPrototips.API/Models/AutoPartsRequestModel.cs
namespace ADASPrototips.API.Models;

public class AutoPartsRequestModel
{
    public int Id { get; set; }

    public AutoPartsRequestStatus Status { get; set; }
    
    public string Justification { get; set; }
    
    public List<AutoPartsRequestComponentModel> Components { get; set; }
}

File: ADASPrototips.Web/react-client-app/src/models/AutoPartsRequestModel.ts
import { AutoPartsRequestStatus } from "./AutoPartsRequestStatus";
import { AutoPartsRequestComponentModel } from "./AutoPartsRequestComponentModel";

export interface AutoPartsRequestModel {
    id: number;
    status: AutoPartsRequestStatus;
    justification: string;
    components: AutoPartsRequestComponentModel[];
}